pyworkflow/pyworkflow/workflow.py

```diff
@@ -565,7 +587,13 @@ def execute_workflow(workflow_location):
         #execute each node in the order returned by execution order method
         #TODO exception handling: stop and provide details on which node failed to execute
         for node in execution_order:
-            workflow_instance.execute(node)
+            if type(workflow_instance.get_node(node)) is ReadCsvNode and len(stdin_files) > 0:
+                csv_location = stdin_files[0]
+                workflow_instance.execute_read_csv(node, csv_location)
+                # delete file at index 0
+                del stdin_files[0]
+            else:
+                workflow_instance.execute(node)
```

Comment:
I ran into this bug the other day where the graph didn't update after execution. The `execute` call returns the executed node for the front-end to then call `workflow.update_or_add_node()` to actually store the `node.data` attribute. Without this, I got complaints that predecessor data was missing when it's actually written to disk.

Changing both if/else to include `executed_node = workflow_instance.<execute_method_here>...` and then `workflow_instance.update_or_add_node(executed_node)` outside the if/else but within the for loop solves the issue. Come to think of it, we should probably update the `execute` method to have this behavior and update the execute endpoint as well (to avoid double-saving).

Reply:
I see, I've modified the code to do exactly what you suggested and seems to be working fine.

Reply:
I've added a modification that includes the dataframe output in sdtout. Didn't need to duplicate the exact functionality as ReadCsv I was able to print output = node_to_execute.execute(preceding_data, execution_options). I'm still not happy with the repeating of execute code in workflow, so will be refactoring this in the next PR.